### PR TITLE
Add OTel Operator status section

### PR DESCRIPTION
**Pull request by cartermp**
fixes https://github.com/open-telemetry/opentelemetry.io/issues/4131

@jaronoff97 Would love some help finding the right word to define the actual operator status. Is it stable/beta/alpha/etc?

### CODE REVIEW

**Review by svrnm — CHANGES_REQUESTED** [review]

File: content/en/status.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ core collector components currently have mixed
 stability documented in its `README.md`. You can find a list of all available
 collector components in the [registry](/ecosystem/registry/?language=collector).
 
+## Kubernetes Operator
+
+The OpenTelemetry Operator status is
+[mixed](/docs/specs/otel/document-status/#mixed), since it can deploy components
+of differing statuses.
+
+The Operator itself is considered TODO LOL.
```
[the remark this answers is not in the document]
Comment (theletterf) [reply content/en/status.md:45]: I kind of like `TODO LOL` as a maturity level.

[the remark this answers is not in the document]
Comment (pavolloffay) [reply content/en/status.md:45]: I think we should talk about statuses of the  CRDs managed by the operator (`OpenTelemetryCollector` and `Instrumentation`, `OpAMPBridge`).

[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/status.md:45]: That's a good idea. What would the statuses be?

[the remark this answers is not in the document]
Comment (svrnm) [reply content/en/status.md:45]: @pavolloffay can you help @cartermp to update the TODO?

[the remark this answers is not in the document]
Comment (pavolloffay) [reply content/en/status.md:45]: The fields in the CRD exposed by the operator should not change https://github.com/open-telemetry/opentelemetry-operator/tree/main/apis. At the moment there are under v1alpha1 and v1beta1

[the remark this answers is not in the document]
Comment (cartermp) [reply content/en/status.md]: Thanks @pavolloffay, I updated with this info.

**Comment by cartermp** [comment]
@svrnm please give this another look!

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@open-telemetry/operator-approvers PTAL!

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**